zebra-consensus/src/verify.rs

```diff
@@ -52,24 +51,26 @@ impl Service<Block> for ZebraBlock {
     fn call(&mut self, block: Block) -> Self::Future {
         let mut state_service = zebra_state::in_memory::init();
 
-        // TODO(teor): error handling?
-        state_service.call(zebra_state::Request::AddBlock {
+        let header_hash: BlockHeaderHash = (&block).into();
+
+        // Ignore errors, they are already handled correctly.
```

Comment:
It's not that we're ignoring errors, its just that there are none.

Reply:
I think I might be confused by the design here. (Or the terminology.)

ZebraState rejects blocks if another block is added at the same height. (The block can be the same as the existing block, or a chain fork.)
https://github.com/ZcashFoundation/zebra/blob/main/zebra-state/src/in_memory/block_index.rs#L31

If we make BlockVerifier satisfy the ZebraState preconditions, then there will be no errors here. But I haven't done that yet.

How should BlockVerifier handle `Err("forks in the chain aren't supported yet")` ?

Reply:
oh, my bad I misunderstood what this was doing. I didn't realize that you're just discarding the result, yea error handling will be necessary, we will probably have to switch to using real error types so we can handle them effectively.

Reply:
Ok, I'll mark that down as your TODO?

Reply:
I split the TODOs between us both in 74b6cc790379fa50d7bf24a238aa5b556a3d6f9d.